### PR TITLE
93 enhance page header social share panel

### PR DESCRIPTION
**Pull request by Khalil-NOUI**
-enable tab title meta Info and enable share feature in the single event page
-enhance translation in some lacking components

### CODE REVIEW

File: src/components/EditProfileComponenets/updateSignInDetails.jsx [hunk 1]
```diff
@@ -129,112 +131,116 @@ function UpdateSignInDetailes() {
...
-                        minLength='8'
-                    />
-                </div>
-                {error && (
-                    <div className='flex justify-center items-center'>
-                        <p className='text-red-500 border-white w-fit bg-red-300 rounded-lg slide-in'>
-                            {error}
-                        </p>
+        <>
+            <IndexPage title='Edit Profile' />
```
Comment (HachemBouhadede) [inline src/components/EditProfileComponenets/updateSignInDetails.jsx:135]: shouldn't this be in the profile/[id]/editProfile ? it shouldn't be in componenets

File: src/components/singlEventPageComponents/EventDetails.jsx [hunk 2]
```diff
@@ -226,368 +203,365 @@ function EventDetails() {
...
-                            <div
-                                onClick={() => {
-                                    navigateToEvent("next");
-                                }}
-                                className='absolute -right-20 top-1/3 h-28 w-7 cursor-pointer group-hover:right-2 transition-all duration-300 flex justify-center items-center text-2xl bg-white hover:bg-gray-200 rounded-md opacity-75 text-black hover:text-green-500  z-10'
-                            >
-                                <HiChevronRight />
-                            </div>
+        <>
+            <IndexPage title={`${eventDisplay.title}`} />
```
Comment (HachemBouhadede) [inline src/components/singlEventPageComponents/EventDetails.jsx:210]: same headr, i supposed headers go to pages not componenets

File: src/components/singlEventPageComponents/EditEvent.jsx [hunk 3]
```diff
@@ -156,144 +158,161 @@ function EditEvent() {
...
-                            height={60}
-                        ></Image>
-                        <h1 className='text-4xl font-bold text-center mb-2 font-Montserrat text-txtc-Ivory'>
-                            {t("editEvent.title")}
-                        </h1>
-                        <p className='text-lg font-Lora text-center text-txtc-Ivory'>
-                            {t("editEvent.intro")}
-                        </p>
+        <>
+            <IndexPage title={`Edit Event ${oldInfo.title}`} />
```
Comment (HachemBouhadede) [inline src/components/singlEventPageComponents/EditEvent.jsx:162]: same here

File: src/components/reusableComponents/indexPage.jsx
Comment (HachemBouhadede) [inline src/components/reusableComponents/indexPage.jsx:1]: i get the logic of writing as a component, i think it's actually brilliant, but with all these changes, i think the standard way could've easier, anyway, hope this gets solved out soon ! and great efforts khalil

File: src/components/eventsPageComponents/EventList.jsx [hunk 4]
```diff
@@ -202,193 +204,198 @@ function EventList() {
...
-                <div className='hidden bg-gray-50 mt-4 sticky top-24 rounded-xl shadow-lg md:flex flex-col w-1/4 gap-4  h-[60%]'>
-                    {/* reset filter button */}
-                    <div
-                        className='flex flex-row mt-3 h-auto py-1 justify-between px-1 transition-all duration-150 text-lg bg-gray-200 font-medium cursor-pointer hover:text-white items-center w-auto gap-2 self-center hover:bg-red-500 rounded-full shadow-lg'
-                        onClick={() => resetFilter()}
-                    >
-                        <div className='w-7 h-7 bg-gray-50 rounded-full text-red-500 flex justify-center items-center'>
-                            <LuFilterX />
+        <>
+            <IndexPage title='All Events' />
```
Comment (HachemBouhadede) [inline src/components/eventsPageComponents/EventList.jsx:208]: Aide for Aide Events or Our Events ! could be nice :D

File: src/components/createEventPageComponents/CreateEvent.jsx [hunk 5]
```diff
@@ -86,142 +88,157 @@ function CreateEvent() {
...
-                            height={60}
-                        ></Image>
-                        <h1 className='text-4xl font-bold text-center mb-2 font-Montserrat text-white'>
-                            {t("createEvent.title")}
-                        </h1>
-                        <p className='text-lg font-Lora text-center text-white'>
-                            {t("createEvent.intro")}
-                        </p>
+        <>
+            <IndexPage title='Create an Event' />
```
Comment (HachemBouhadede) [inline src/components/createEventPageComponents/CreateEvent.jsx:92]: i kinda unerstand why you'd use the indexPage components in componenets that are meant for dynamic pages, but for static pages, i defently think it's prefarble to put them in pages

File: src/pages/profile/[id]/events/index.jsx [hunk 6]
```diff
@@ -54,74 +55,78 @@ function Events() {
...
-                                <Image
-                                    src='/images/defaultUser.png'
-                                    alt='Default'
-                                    className='rounded-full border-orange-400 border-2 w-24 h-24 lg:w-32 lg:h-32 xl:w-40 xl:h-40'
-                                    width={120}
-                                    height={120}
-                                />
-                            )}
+        <>
+            <IndexPage title={`${userData.displayName} events`} />
```
Comment (HachemBouhadede) [inline src/pages/profile/[id]/events/index.jsx:59]: yeah this seems more logical

File: src/pages/index.jsx [hunk 7]
```diff
@@ -15,6 +16,7 @@ export default function HomePage({ props }) {
 
     return (
         <>
+            <IndexPage title='Pebble' />
```
Comment (HachemBouhadede) [inline src/pages/index.jsx:19]: don't forget to update the title

File: src/components/singlEventPageComponents/EventDetails.jsx (hunk 2 above)
Comment (HachemBouhadede) [reply src/components/singlEventPageComponents/EventDetails.jsx:210]: but i understand why you're using it her, anyway, as long as it works <3

**Review by HachemBouhadede — COMMENTED** [review]
let's not forget to delete all unceccessary comment and console.logs ! hope this gets solved soon

**Comment by HachemBouhadede** [comment]
image

current error, 
i did a quick search and found that it's okey to use a header inside nested componenets instead of a page; so that theoretically isn't an issue, but you never know, if you get no luck try to reach to a trainer

**Comment by Khalil-NOUI** [comment]
@HachemBouhadede
  Thank you,Hacheim for the feedback, I'll address of it as much as i can in the next push

File: src/pages/index.jsx (hunk 7 above)
Comment (Khalil-NOUI) [reply src/pages/index.jsx:19]: Aide be Aide  :)

**Comment by HachemBouhadede** [comment]
still same issue with vercel
image


here's a suggution idea @Khalil-NOUI  for this page make the header a normal one don't use the indexPage componenet @Khalil-NOUI  just to solve it out

**Comment by Khalil-NOUI** [comment]
All good to GO !,

**Review by HachemBouhadede — APPROVED** [review]

**Pull request merged**